File: 문민우/프로그래머스/level2/의상.java
package level2;

import java.util.Map;
import java.util.HashMap;

public class 의상 {
    public int solution(String[][] clothes) {

        int answer = 1;

        HashMap<String, Integer> clothingMap = new HashMap<>();

        for (String[] clothing : clothes) {

            if (clothingMap.containsKey(clothing[1]))
                clothingMap.put(clothing[1], clothingMap.get(clothing[1]) + 1);

            else
                clothingMap.put(clothing[1], 2);
        }

        for (String clothing : clothingMap.keySet()) {
            answer *= clothingMap.get(clothing);
        }

        answer--;

        return answer;
    }
}

File: 이솔/프로그래머스/Level2/의상.py
def solution(clothes):
    clo={}
    for i in clothes:
        item=i[0]
        cate=i[1]
        
        if cate in clo:
            clo[cate].append(item)
        else:
            clo[cate] = [item]

    answer=1
    for key in clo.keys():
        answer = answer * (len(clo[key]) + 1)
    return answer - 1
